omnigibson/object_states/particle_modifier.py

```diff
@@ -1193,6 +1205,11 @@ def _get_max_particles_limit_per_step(self, system):
         return m.MAX_VISUAL_PARTICLES_APPLIED_PER_STEP if is_visual_particle_system(system_name=system.name) else \
             m.MAX_PHYSICAL_PARTICLES_APPLIED_PER_STEP
 
+    @property
+    def requires_overlap(self):
+        # Overlap required only if sampling with raycast
+        return self._sample_with_raycast
```

Comment:
nit: should we just unify `sample_with_raycast` with `requires_overlap`? Seems weird that we have two different names referring to the exact same thing.

Reply:
yeah, good point. I think semantically it's a bit more clear though with these two terms -- `requires_overlap` applies to all ParticleModifiers, but `sample_with_raycast` is specific to `ParticleAppliers`, since we're actually sampling particle positions if that is true.

I'm a bit ambivalent though, so don't mind changing it too.

Reply:
Got it. Let's just keep the status quo then.